src/nnet3bin/nnet3-copy-egs.cc

```diff
@@ -316,28 +369,60 @@ int main(int argc, char *argv[]) {
 
     SequentialNnetExampleReader example_reader(examples_rspecifier);
 
+    RandomAccessTokenReader output_reader(eg_output_rspecifier);
+    RandomAccessBaseFloatReader egs_weight_reader(eg_weight_rspecifier);
     int32 num_outputs = po.NumArgs() - 1;
     std::vector<NnetExampleWriter*> example_writers(num_outputs);
     for (int32 i = 0; i < num_outputs; i++)
       example_writers[i] = new NnetExampleWriter(po.GetArg(i+2));
 
 
-    int64 num_read = 0, num_written = 0;
+    int64 num_read = 0, num_written = 0, num_err = 0;
     for (; !example_reader.Done(); example_reader.Next(), num_read++) {
       // count is normally 1; could be 0, or possibly >1.
       int32 count = GetCount(keep_proportion);
       std::string key = example_reader.Key();
-      const NnetExample &eg = example_reader.Value();
+      bool modify_eg_output = !(eg_output_rspecifier.empty() &&
+                                eg_weight_rspecifier.empty());
+      NnetExample eg_modified_output;
+      const NnetExample &eg_orig = example_reader.Value(),
+        &eg = (modify_eg_output ? eg_modified_output : eg_orig);
+      // rename io name 'output' to output-name and scale supervision using weight.
```

Comment:
this comment is confusing, please remove it and comment:
// Note: in the normal case we just use 'eg'; eg_modified_output is
// for the case when the --outputs or --weights option is specified
// (only for multilingual training).